r/R/query-engine.R

```diff
@@ -22,7 +22,12 @@ do_exec_plan <- function(.data) {
 
   # TODO (ARROW-14289): make the head/tail methods return RBR not Table
   if (inherits(tab, "RecordBatchReader")) {
-    tab <- tab$read_table()
+    tryCatch(
+      tab <- tab$read_table(),
+      error = function(e) {
+        handle_csv_read_error(e, tab$schema)
```

Comment:
I think you want this check higher up, in `collect.arrow_dplyr_query`, around the `do_exec_plan()` call. It feels fragile and out of place here.